raiden/network/pathfinding.py

```diff
@@ -264,3 +302,51 @@ def query_paths(
             'Pathfinding service returned invalid json',
             dict(response_text=response.text, exc_info=True),
         )
+
+
+def query_paths(
+        service_config: typing.Dict[str, typing.Any],
+        our_address: typing.Address,
+        privkey: bytes,
+        current_block_number: typing.BlockNumber,
+        token_network_address: typing.Union[typing.TokenNetworkAddress, typing.TokenNetworkID],
+        route_from: typing.InitiatorAddress,
+        route_to: typing.TargetAddress,
+        value: typing.TokenAmount,
+) -> typing.List[typing.Dict[str, typing.Any]]:
+
+    max_paths = service_config['pathfinding_max_paths']
+    url = service_config['pathfinding_service_address']
+    payload = {'from': route_from, 'to': route_to, 'value': value, 'max_paths': max_paths}
+    offered_fee = service_config.get('pathfinding_fee', service_config['pathfinding_max_fee'])
+    scrap_existing_iou = False
+
+    for retries in reversed(range(MAX_PATHS_QUERY_ATTEMPTS)):
+        payload.update(create_current_iou(
+            config=service_config,
+            token_network_address=token_network_address,
+            our_address=our_address,
+            privkey=privkey,
+            block_number=current_block_number,
+            offered_fee=offered_fee,
+            scrap_existing_iou=scrap_existing_iou,
+        ))
+
+        try:
+            return post_pfs_paths(
+                url=url,
+                token_network_address=token_network_address,
+                payload=payload,
+            )
+        except ServiceRequestIOURejected as error:
+            code = error.error_code
+            if retries == 0 or code in (PfsError.WRONG_IOU_RECIPIENT, PfsError.DEPOSIT_TOO_LOW):
+                raise
```

Comment:
This would re-raise the exception... but where do we handle this outside the routing module? I think it would make sense to handle this exception by:
1. Either return an error back to the API that called initiating a transfer
2. Fall back to the basic routing module

@hackaugusto thoughts on this?

Reply:
See `routing.py:170`. All unhandled exceptions fall through to `get_best_routes_pfs` and are handled there. Currently we log a warning that the PFS query has failed, and return an empty list of route states, which will cause the transfer to fail with an error response in the API.